aiida_cp2k/utils/input_generator.py

```diff
@@ -207,9 +207,19 @@ def add_wfn_restart_section(input_dict, is_kpoints):
 
 
 @calcfunction
-def add_ext_restart_section(input_dict):
+def add_ext_restart_section(input_dict, first_snapshot=None):
     """Add external restart section to the input dictionary."""
     params = input_dict.get_dict()
+    if first_snapshot is not None:
+        params["MOTION"]["MD"]["REFTRAJ"]["FIRST_SNAPSHOT"] = first_snapshot
     # overwrite the complete EXT_RESTART section if present
     params["EXT_RESTART"] = {"RESTART_FILE_NAME": "./parent_calc/aiida-1.restart"}
+    if params["GLOBAL"]["RUN_TYPE"] == "MD":
```

Comment:
Should we not just enable it always?

Reply:
what whould you like to enable always? FIRST_SNAPSHOT exist only in the REFTRAJ ensemble, it is not a general MD feature

Reply:
For the moment, we agreed to keep them all set explicitly. If an issue pops up, we will revisit this section.

Especially, we need to monitor the replica chain calculations, which use constraints.